### DISCUSSION TITLE
Use case for Pyjion

### DISCUSSION THREAD

**Discussion by thomas-t1** [tonybaloney/Pyjion#303] [Q&A] [answered]
I'm trying to understand possible uses of Pyjion, I came across the project while looking for a way to host python code easily along side a .net 5 application.

My question is if there is any possibility in the future to generate a CIL assembly containing the python program/code generated and be able to execute it through the .NET CLR without the need for CPython? E.g. to distribute hybrid .Net/Python applications.

I understand that there would be no direct interaction between CIL python and e.g. C#, but this could be solved by other means.

**Comment by tonybaloney** [comment, marked as answer]
I don't have any plans to do this. It would mean exposing the Python Object/Type system via the typed CIL instructions. I think this would be a lot of work.

There is also the IronPython project, which does this.
